python/dgl/nn/pytorch/explain/subgraphxexplainer.py

```diff
@@ -0,0 +1,566 @@
+
+    marginal_contributions = torch.cat(marginal_contribution_list, dim=0)
+    return marginal_contributions
+
+
+def shapley(
+    model, graph, subgraph_nodes, num_gnn_layers, mc_sampling_steps, feat
+):
+    r"""Monte carlo sampling approximation of the shapley value.
+
+    Parameters
+    ----------
+    model: nn.Module
+        The GNN model to explain.
+    graph: DGLGraph
+        A homogeneous graph.
+    subgraph_nodes: tensor
+        The tensor node ids of the subgraph that are associated with this tree node.
+    num_gnn_layers: int
+        Number of layers of GNN model. Needed to calculate the l-hop neighbouring
+        nodes of graph. Computing shapley values for big and complex graphs
+        is time and resource consuming. This algorithm efficiently approximate
+        shapley values by finding coalition of l-hop neighboring nodes of the graph.
+        The GNN model has l layers.
+    mc_sampling_steps: int
+        Monte carlo sampling steps.
+    feat: Tensor
+        The input feature of shape :math:`(N, D)`. :math:`N` is the
+        number of nodes, and :math:`D` is the feature size.
+
+    Returns
+    -------
+    float
+        Returns the shapley value based on the subgraph nodes.
+    """
+
+    num_nodes = graph.num_nodes()
+
+    local_region = subgraph_nodes.tolist()
+    for _ in range(num_gnn_layers - 1):
+        k_neighbourhood = []
+        for node in local_region:
+            k_neighbourhood += set(
+                graph.successors(node).tolist()
+                + graph.predecessors(node).tolist()
+            )
+        local_region += k_neighbourhood
+        local_region = list(set(local_region))
+
+    coalition = subgraph_nodes.tolist()
+    coalition_placeholder = num_nodes
+    set_exclude_masks = []
+    set_include_masks = []
+    for _ in range(mc_sampling_steps):
+        subset_nodes_from = list(set(local_region) - set(subgraph_nodes))
+        random_nodes_permutation = subset_nodes_from + [coalition_placeholder]
+        random_nodes_permutation = np.random.permutation(
+            random_nodes_permutation
+        )
+        split_idx = np.where(random_nodes_permutation == coalition_placeholder)[
+            0
+        ][0]
+
+        selected_nodes = random_nodes_permutation[:split_idx]
+
+        set_exclude_mask = np.ones(num_nodes, dtype=np.int8)
+        set_exclude_mask[local_region] = 0.0
+        set_exclude_mask[selected_nodes] = 1.0
+        set_include_mask = set_exclude_mask.copy()
+        set_include_mask[coalition] = 1.0
+
+        set_exclude_masks.append(set_exclude_mask)
+        set_include_masks.append(set_include_mask)
+
+    exclude_masks = np.stack(set_exclude_masks, axis=0)
+    include_masks = np.stack(set_include_masks, axis=0)
+    marginal_contributions = marginal_contribution(
+        graph, exclude_masks, include_masks, model, feat
+    )
+
+    shapley_value = marginal_contributions.mean().item()
+    return shapley_value
+
+
+class MCTSNode:
```

Comment:
No need to pass `high2low` and `coef` to `MCTSNode`

Reply:
coef is needed to get the action selection criteria.